utilities/additionalProperties.go

```diff
@@ -1,34 +1,68 @@
 package utilities
 
-import "encoding/json"
+import (
+	"encoding/json"
+	"errors"
+	"strings"
+)
+
+func ValidateAdditionalProperty[T any](dstMap map[string]T, keysToRemove ...string) error {
+	containsKey := func(key string) bool {
+		for _, tag := range keysToRemove {
+			if tag == key {
+				return true
+			}
+		}
+		return false
+	}
+
+	for key, _ := range dstMap {
+		if strings.TrimSpace(key) == "" {
+			return errors.New("an additional property key can not be empty or whitespace")
+		}
+		if containsKey(key) {
+			return errors.New("an additional property key, '" + key + "' conflicts with one of the model's properties")
+		}
+	}
+	return nil
+}
 
 // MapAdditionalProperties append additional properties to destination struct map
-func MapAdditionalProperties(destinationMap additionalProperties, sourceMap additionalProperties) {
-	destinationMap.appendMap(sourceMap)
+var MapAdditionalProperties = MapAdditionalProperty[any]
```

Comment:
Are we only using this in the TestMapAdditionalProperties only?

Reply:
I have added it to avoid breaking changes for existing users.